types/filter.d.ts

```diff
@@ -10,11 +10,16 @@ export function filter<A, P extends A>(
   <B extends A>(list: readonly B[]): P[];
 };
 
-// filter(() => boolean)
+// filter(() => boolean, list)
 export function filter<T>(
   pred: (value: T) => boolean,
 ): <P extends T, C extends readonly P[] | Record<string, P>>(collection: C) => C;
```

Comment:
You need to add support for `Map` here as an overload, similar to how `filter(() => narrow)` is above. You need to add it up there too as well, actually

```suggestion
export function filter<T>(
  pred: (value: T) => boolean,
): {
  <K>(map: Map<K, T>): Map<K, T>;
  <P extends T, C extends readonly P[] | Record<string, P>>(collection: C): C;
};
```